PythonRegression/tests/features/steps/api_test_steps.py

```diff
@@ -1,21 +1,66 @@
 from aloe import *
-from iota import Iota
+from iota import Iota, ProposedTransaction, Address
 from iota.commands.core import add_neighbors
 from util import static_vals
+from util.test_logic import api_test_logic
+from yaml import load, Loader
+
+import logging 
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+tests = api_test_logic
 
 neighbors = static_vals.TEST_NEIGHBORS
 testHash = static_vals.TEST_HASH
 testTrytes = static_vals.TEST_TRYTES
 
 config = {}
-responses = {'getNodeInfo':{},'getNeighbors':{},'getTips':{},'getTrytes':{}}   
+responses = {'getNodeInfo':{},'getNeighbors':{},'getTips':{},'getTransactionsToApprove':{},'getTrytes':{}}   
+
+yamlPaths = ["./tests/features/machine1/output.yml","./tests/features/machine2/output.yml",
+             "./tests/features/machine3/output.yml"]
+
+#Configuration
+@before.all
+def configuration():
+    logger.info('Start Node Configuration')
+    machines = {}  
+    count = 1  
+    for i in range(len(yamlPaths)):
+        stream = open(yamlPaths[i],'r')
+        machine = load(stream,Loader=Loader)
+        world.seeds = machine.get('seeds')  
+        nodes_final = {}
+        keys = machine.keys()   
+        logger.debug('Keys: %s',keys) 
+        for i in keys:
+            if i == 'nodes':
+                for x in machine[i]:
+                    name = x
+                    host = machine[i][x]['host']
+                    port = machine[i][x]['ports']['api']
+                    udpPort = machine[i][x]['ports']['gossip-udp']
+                    logger.info(host + ":" + str(port))
+                    nodes_final[name] = {'host': host, 'port': str(port), 'udpPort': str(udpPort)}
+                    logger.info('%s configured', name)
+
+        machine_tag = 'machine'+str(count)
+        machines[machine_tag] = nodes_final
+        logger.info('Machine %d configured', count)  
+        count += 1
+
+              
+    world.machines = machines
+    logger.info('Node Configuration Complete')
 
 ###
 #Register API call    
 @step(r'"([^"]*)" is called on "([^"]*)"')
```

Comment:
@karimodm in each feature file that is calling the step parameters will be a bit different. This may also be confusing. You can understand what the regex is from the method signature.

You agree?

Reply:
I would like at least a comment to make sure this is an API call